apps/sms/test/marionette/notification_test.js

```diff
@@ -225,7 +226,63 @@ marionette('Message notification tests', function() {
 
       openNotification();
 
-      assertMessagesIsInCorrectState();
+      assertMessagesIsInCorrectStateFromMessage(smsMessage);
+    });
+  });
+
+  suite('Display a conversation from a notification', function() {
+    var messagesStorage;
+    var thread1, thread2;
+
+    setup(function() {
+      thread1 = ThreadGenerator.generate({
+        numberOfMessages: 50,
+        participants: ['999']
+      });
+      thread2 = ThreadGenerator.generate({
+        numberOfMessages: 50,
+        participants: ['888']
+      });
+
+      messagesStorage = [ thread1, thread2 ];
+    });
+
+    test('Clicking a notification while a conversation is loading', function() {
+      messagesApp.launch();
+      storage.setMessagesStorage(messagesStorage);
+      // Set empty contacts store.
+      storage.setContactsStorage();
+
+      var inbox = new InboxView(client);
+      var conversation = inbox.goToConversation(
+        thread1.id, { shouldWait: false }
```

Comment:
ok, I tried with much more messages in the first thread, and you're right it reproduces the problem now. I'll revert the "shouldWait" changes.